### PR TITLE
Migrate all FO pages to TS

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Migrate all FO pages to TS
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | Nightly is green => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/3847727629
| Possible impacts? | no

### CODE REVIEW

**Review by Progi1984 — COMMENTED** [review]
Seems a file is missing : `tests/UI/pages/FO/pricesDrop/index.ts` ?

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
@nesrineabdmouleh Your CI is :red_circle:

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
:warning: CI on orders are still failing

**Comment by Progi1984** [comment]
CI is now :green_circle:.

Ping @NeOMakinG.

File: tests/UI/pages/FO/myAccount/vouchers.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+   * @param row {number} Row number in vouchers table
+   * @param columnName {string} Column name in vouchers table
+   * @returns {Promise<string>}
+   */
+  async getTextColumnFromTableVouchers(page: Page, row: number, columnName: string): Promise<string> {
+    let columnSelector:string;
+
+    switch (columnName) {
+      case 'code':
+        columnSelector = this.tableColumnCode(row) as string;
```
Comment (NeOMakinG) [inline tests/UI/pages/FO/myAccount/vouchers.ts:74]: It definitely shows that `tableColumnCode` return type is wrong if you cast it everywhere, and if it's not migrated yet, you'll forget to change these lines :(

**Review by NeOMakinG — APPROVED** [review]
I sent a comment, casts are bad, overwise looks good to me

**Pull request merged**